framework/src/org/checkerframework/framework/type/QualifierHierarchy.java

```diff
@@ -564,15 +565,28 @@ public AnnotationMirror greatestLowerBound(
         }
     }
 
+    /**
+     *
+     * @deprecated use {@link #getCorrespondingAnnotation(Collection, AnnotationMirror)} instead
+     */
+    @Deprecated
     public AnnotationMirror findCorrespondingAnnotation(
             AnnotationMirror aliased, Collection<? extends AnnotationMirror> a) {
-        AnnotationMirror top = this.getTopAnnotation(aliased);
-        for (AnnotationMirror anno : a) {
-            if (this.isSubtype(anno, top)) {
-                return anno;
-            }
-        }
-        return null;
+        return getCorrespondingAnnotation(a, aliased);
+    }
+
+    /**
+     * Returns the annotation from the hierarchy that corresponds to {@code annotationMirror} from
```

Comment:
Hard to parse.   I would reorder to something like:

Returns the annotation that corresponds to the set annos.
The returned annotations will be in the same hierarchy as annotationMirror.

I presume it's an invariant that all the annotations in the set, and the second argument, must be from the same hierarchy, and the second argument is only necessary because the set might be empty.  (Actually, can the set be empty?  If not, what is the point of the second argument?  Are the elements in the set guaranteed to all be in different hierarchies?  Is the set required to be non-empty?